### PR TITLE
Add dev check to API calls

### LINKED ISSUE #197: Add dev flag to disable auth from requests
### Description of the issue

The flag should remove the auth tokens from fia-api requests. This is for local development, to prevent issues with needing to authenticate locally.

### LINKED ISSUE #249: Add script for running frontend + SciGateway in one
### Description of the issue

For ease of development add this script.

### PR DESCRIPTION
**Pull request by Dagonite**
Closes #249, closes #197.

## Description

Adds a dev check to the reduction fetch request. If true, an auth token isn't generated.

Also adds a helpful developer script.

### CODE REVIEW

File: .vscode/tasks.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{
```
Comment (keiranjprice101) [inline .vscode/tasks.json:1]: Do we want ide specific files in the source?
@Pasarus

File: src/ReductionHistory.tsx [hunk 2]
```diff
@@ -105,13 +105,21 @@ const ReductionHistory: React.FC = () => {
 
   const fetchReductions = useCallback(async (): Promise<void> => {
```
Comment (keiranjprice101) [inline src/ReductionHistory.tsx:106]: I believe there is at least one more fetch this would be needed on?

**Review by keiranjprice101 — CHANGES_REQUESTED** [review]
The way of switching makes sense to me.

File: .vscode/tasks.json (hunk 1 above)
Comment (Dagonite) [reply .vscode/tasks.json:1]: I've seen some repos have it. Can we helpful for others developing on VS Code in the future.

File: src/ReductionHistory.tsx (hunk 2 above)
Comment (Dagonite) [reply src/ReductionHistory.tsx:106]: @keiranjprice101 there is in another branch I'm working on but this branch is off of main.

Comment (keiranjprice101) [reply src/ReductionHistory.tsx:106]: Is there not one in the value editor?

**Review by keiranjprice101 — APPROVED** [review]

**Pull request merged**